dt-backend/src/model/user/user.service.ts

```diff
@@ -12,4 +24,37 @@ export class UserService {
       },
     });
   }
+
+  async createUser(data: Prisma.UserCreateInput): Promise<User> {
+    return prisma.user.create({
+      data,
+      include: {
+        audit: true,
+        empire: true,
+      },
+    })
+  }
+
+  async updateUser(params: {where: Prisma.UserWhereUniqueInput; data: Prisma.UserUpdateInput;}): Promise<User> {
```

Comment:
Yeah, here it is - you should just pass through the Controller Object as it is or split it in the controller completely!
Not do it partially as it is atm. So the two ways to do it are either:
`async updateUser(data: UserUpdateDTO): Promise<User> {`
or
`async updateUser(id: number, username: string, ...): Promise<User> {`

I much prefer the first, since it allows us to better document our REST API using OpenAPI Annotations!

Reply:
okay, i created a UserUpdateDTO, but the only concern is that prisma requires both a data field AND a where field. I included them in the dto for brevity though. I will push this soon as well.